tests/HubRestAPIClient.ts

```diff
@@ -0,0 +1,806 @@
+import chai, { expect } from "chai";
+import chaiAsPromised from "chai-as-promised";
+import OpenAPIResponseValidator from "openapi-response-validator";
+import { promises as fs } from "fs";
+import yaml from "yaml";
+import { OpenAPIV3 } from "openapi-types"; // cspell:disable-line
+import type { OverrideProperties } from "type-fest";
+
+import { HubRestAPIClient } from "../src/hub-rest-client";
+import { Logger, silentLogger } from "../src/common/logger";
+import { expectDefinedNonNull } from "./utils";
+import {
+  GetUserDataByFid200ResponseOneOf,
+  LinkType,
+  ListOnChainSignersByFid200ResponseOneOf,
+  OnChainEventType,
+  ReactionType,
+  UserDataType,
+} from "../src/hub-rest-client/openapi";
+import type { paths as SchemaPaths } from "./hub-rest-client/openapi/schema";
+
+chai.use(chaiAsPromised);
+
+const userGaviFid = 69; // @gavi
+
+const testLogger: Logger = {
+  info: silentLogger.info,
+  debug: silentLogger.debug,
+  trace: silentLogger.trace,
+  warn: silentLogger.warn,
+
+  /* eslint-disable no-console */
+  error: console.error,
+  /* eslint-enable no-console */
+};
+
+describe("HubWebClient", function () {
+  let client: HubRestAPIClient;
+  let apiSpec: OverrideProperties<OpenAPIV3.Document, { paths: SchemaPaths }>; // cspell:disable-line
+
+  before("setup", async function () {
+    client = new HubRestAPIClient({
+      server: "http://nemes.farcaster.xyz:2281",
```

Comment:
why aren't we using the sc hub for the tests?

Reply:
We haven't exposed port 2281 on our hub yet. Want to make sure we're deliberate about cost-planning and HTTPS and all that before we open that one up